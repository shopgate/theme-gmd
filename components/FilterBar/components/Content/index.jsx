/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { compareObjects } from '@shopgate/pwa-common/helpers/redux';
import Sort from './components/Sort';
import ViewSwitch from './components/ViewSwitch';
import FilterButton from './components/FilterButton';
import FilterChips from './components/FilterChips';
import connect from './connector';
import styles from './style';

/**
 * The Filter bar component.
 */
class Content extends Component {
  static propTypes = {
    componentUpdated: PropTypes.func.isRequired,
    getFilters: PropTypes.func.isRequired,
    activeFilters: PropTypes.shape(),
  };

  static defaultProps = {
    activeFilters: {},
  };

  static contextTypes = {
    i18n: PropTypes.func,
  };

  /**
   * Called after mount. Sets up the scroll DOM elements.
   */
  componentDidMount() {
    this.props.getFilters();
  }

  /**
   * Called before the component receives new properties. Sets up the scroll DOM elements.
   * @param {Object} nextProps The next component props.
   */
  componentWillReceiveProps(nextProps) {
    if (!compareObjects(nextProps.activeFilters, this.props.activeFilters)) {
      this.props.getFilters();
    }
  }

  /**
   * Call parents' callback to trigger some re-calculation.
   */
  componentDidUpdate() {
    this.props.componentUpdated();
  }

  /**
   * Returns the currency.
   * @return {string}
   */
  get currency() {
    /**
     * TODO: Remove context translation for currency and,
     * instead, get it from shop settings when available.
     */
    const { __ } = this.context.i18n();
    return __('price.currency');
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    return (
      <section>
        <div className={styles}>
          <ViewSwitch />
          <Sort />
          <FilterButton />
        </div>
        <FilterChips currency={this.currency} />
      </section>
    );
  }
}

export const UnwrappedFilterBarContent = Content;

export default connect(Content);
