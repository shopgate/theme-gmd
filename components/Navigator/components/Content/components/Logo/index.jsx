/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import appConfig from '@shopgate/pwa-common/helpers/config';
import styles from './style';

/**
 * The navigator logo component.
 * @return {JSX}
 */
const Logo = () => <img className={styles} src={appConfig.logo} alt="Logo" />;

export default Logo;
