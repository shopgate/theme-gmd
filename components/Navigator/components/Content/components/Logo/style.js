/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

export default css({
  margin: '0 auto',
  maxHeight: '100%',
});
