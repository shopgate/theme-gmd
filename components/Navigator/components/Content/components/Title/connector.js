import { connect } from 'react-redux';
import { getTitle } from '@shopgate/pwa-common/selectors/ui';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  title: getTitle(state),
});

/**
 * @param {Object} next The next state.
 * @param {Object} prev The previous state.
 * @returns {boolean}
 */
const areStatePropsEqual = (next, prev) => (
  prev.title === next.title
);

export default connect(mapStateToProps, null, null, { areStatePropsEqual });
