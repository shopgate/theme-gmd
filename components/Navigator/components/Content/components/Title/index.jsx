import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ACTION_POP } from '@virtuous/conductor/constants';
import getCurrentAction from '@virtuous/conductor-helpers/getCurrentAction';
import I18n from '@shopgate/pwa-common/components/I18n';
import connect from './connector';
import styles from './style';

/**
 * The Navigator Title component.
 */
class Title extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
    onClick: PropTypes.func,
  };

  static defaultProps = {
    onClick: () => {},
  };

  /**
   * The constructor.
   * Sets the initial state from connected props.
   * @param {Object} props The component props.
   */
  constructor(props) {
    super(props);

    this.previousTitle = '';
    this.title = props.title;
  }

  /**
   * When the component receives new props, preserve the current title.
   * @param {Object} nextProps The components next props.
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.title !== '' && nextProps.title !== this.title) {
      this.previousTitle = this.title;
      this.title = nextProps.title;
    }
  }

  /**
   * Determine how the views should be displayed.
   * @returns {JSX}
   */
  get transitionClass() {
    // For the first page don't do any animation.
    if (!this.previousTitle) {
      return {
        inactive: '',
        active: '',
      };
    }

    const pop = getCurrentAction() === ACTION_POP;

    return {
      inactive: pop ? styles.centerToRight : styles.centerToLeft,
      active: pop ? styles.leftToCenter : styles.rightToCenter,
    };
  }

  /**
   * Renders the component.
   * Displays the Title.
   * @returns {JSX}
   */
  render() {
    const transition = this.transitionClass;

    return (
      <div aria-hidden onClick={this.props.onClick}>
        {/* Renders the inactive / previous title */}
        <div className={`${styles.title} ${transition.inactive}`}>
          <I18n.Text string={this.previousTitle} />
        </div>

        {/* Renders the active / current title */}
        <div className={`${styles.title} ${transition.active}`} data-test-id={`title: ${this.props.title}`}>
          <I18n.Text string={this.props.title} />
        </div>
      </div>
    );
  }
}

export default connect(Title);
