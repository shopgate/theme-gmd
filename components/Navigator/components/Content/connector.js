/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { connect } from 'react-redux';
import { getQueryParam } from '@shopgate/pwa-common/selectors/history';
import submitSearch from '../../actions/submitSearch';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  getQueryParam: param => getQueryParam(state, param),
  path: state.history.pathname,
  searchActive: state.navigator.searchActive,
});

/**
 * Connects the dispatch function to a callable function in the props.
 * @param {Function} dispatch The redux dispatch function.
 * @return {Object} The extended component props.
 */
const mapDispatchToProps = dispatch => ({
  submitSearch: () => dispatch(submitSearch()),
});

export default connect(mapStateToProps, mapDispatchToProps);
