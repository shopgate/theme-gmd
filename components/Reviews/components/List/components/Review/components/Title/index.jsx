/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';

/**
 * Review Title Component.
 * @param {string} title The title of the review.
 * @returns {JSX}
 */
const Title = ({ title }) => (
  <div className={styles}>{title}</div>
);

Title.propTypes = {
  title: PropTypes.string,
};

Title.defaultProps = {
  title: null,
};

export default Title;
