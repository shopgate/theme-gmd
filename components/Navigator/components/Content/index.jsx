import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import { INDEX_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
import Logo from './components/Logo';
import Title from './components/Title';
import Search from './components/Search';
import Suggestions from './components/Suggestions';

/**
 * The NavigatorContent component.
 */
class NavigatorContent extends PureComponent {
  static propTypes = {
    routePattern: PropTypes.string,
  };

  static defaultProps = {
    routePattern: null,
  };

  /**
   * @return {boolean}
   */
  get isIndexRoute() {
    return this.props.routePattern === INDEX_PATH;
  }

  /**
   * @return {JSX}
   */
  render() {
    return (
      <Fragment>
        {this.isIndexRoute && <Logo />}
        {!this.isIndexRoute && <Title />}
        <Search />
        <Suggestions />
      </Fragment>
    );
  }
}

export default NavigatorContent;
