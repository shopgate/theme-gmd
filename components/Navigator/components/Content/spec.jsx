/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Content from './index';
import Logo from './components/Logo';
import Title from './components/Title';

// Mock the redux connect() method instead of providing a fake store.
jest.mock('./connector', () => (obj) => {
  const newObj = obj;

  newObj.defaultProps = {
    ...newObj.defaultProps,
    historyLength: 1,
    action: 'POP',
    title: 'My Title',
    submitSearch: () => {},
    getQueryParam: () => null,
  };

  return newObj;
});

describe('<Content />', () => {
  it('should render the logo', () => {
    const wrapper = shallow(
      <Content path="/" />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Logo).exists()).toBe(true);
  });

  it('should render the title', () => {
    const wrapper = shallow(
      <Content path="some/other/path" />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Title).exists()).toBe(true);
  });
});
