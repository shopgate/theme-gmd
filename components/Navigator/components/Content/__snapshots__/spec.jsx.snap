// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render the logo 1`] = `
<React.Fragment>
  <Logo />
  <Connect(Search)
    active={false}
  />
</React.Fragment>
`;

exports[`<Content /> should render the title 1`] = `
<React.Fragment>
  <Connect(Title)
    onClick={[Function]}
  />
  <Connect(Search)
    active={false}
  />
</React.Fragment>
`;
